src/utils/line.ts

```diff
@@ -785,3 +776,45 @@ export const formatOptimisticProposal = (
 
   return proposedPosition;
 };
+
+// TODO: Move code from getLines to create state.categories into this function
+export const formatLineCategories = (
+  lines: { [key: string]: SecuredLine },
+  categories: { [key: string]: string[] }
+): { [key: string]: string[] } => {
+  const validLines = _.filter(lines, (line) => line.status === 'active' && line.end * 1000 > Date.now());
+
+  const highestCreditLines = _.map(
+    _.sortBy(validLines, (line) => Number(BigNumber.from(line.deposit).div(BASE_DECIMALS)))
+      .reverse()
+      .slice(0, 3),
+    'id'
+  );
+
+  const highestCreditLinesCategory = { 'market:featured.highest-credit': highestCreditLines };
+
+  const highestRevenueLines = _.map(
+    _.sortBy(validLines, (line) => Number(BigNumber.from(line!.spigot!.revenueValue).div(BASE_DECIMALS)))
+      .reverse()
+      .slice(0, 3),
+    'id'
+  );
+
+  const highestRevenueLinesCategory = { 'market:featured.highest-revenue': highestRevenueLines };
+
+  const newCreditLines = _.map(
+    _.sortBy(validLines, (line) => line.start),
+    'id'
+  );
+
+  const newCreditLinesCategory = { 'market:featured.newest': newCreditLines };
+
+  const updatedCategories = {
+    ...categories,
+    ...highestCreditLinesCategory,
+    ...highestRevenueLinesCategory,
+    ...newCreditLinesCategory,
+  };
+
+  return updatedCategories;
```

Comment:
Would prefer u define key/value directly here instead of destructuring.

Reply:
yes, this is a lot cleaner and reduces the amount of code. 

done!